clients/src/main/java/org/apache/kafka/clients/producer/MockProducer.java

```diff
@@ -179,10 +179,18 @@ public void beginTransaction() throws ProducerFencedException {
         this.sentOffsets = false;
     }
 
+    @SuppressWarnings("deprecation")
```

Comment:
Shouldn't this have a deprecated annotation instead of the warning suppression?

Reply:
I guess it doesn't need to have either, I can remove in a followup PR